src/create-cache.js

```diff
@@ -4,6 +4,130 @@ import log4js from 'log4js';
 
 const logger = log4js.getLogger('dreamworld.redis-cache.cache-manager');
 
+/**
+ * It adds given keys into `__keys__` Set.
+ * @param {*} cache 
+ * @param {Array} keys 
+ */
+const _setKeys = (cache, keys) => {
+  return new Promise((resolve, reject) => {
+    const redisClient = cache.store.getClient();
+    redisClient.sadd(['__keys__', ...keys], function (err, res) {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      logger.trace(`set: key added into Set. key=${keys}`);
+      resolve();
+    });
+  });
+};
+
+/**
+ * It removes given keys from `__keys__` Set.
+ * @param {*} cache 
+ * @param {Array} keys 
+ */
+const _deleteKeys = (cache, keys) => {
+  return new Promise((resolve, reject) => {
+    const redisClient = cache.store.getClient();
+
+    redisClient.srem(['__keys__', ...keys], function (err, res) {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      logger.trace(`del: keys removed from Set. keys=${keys}`);
+      resolve();
+    });
+  });
+};
+
+/**
+ * Overrides given redisCache's set() method.
+ * @param {*} cache 
+ */
+const overrideSet = (cache) => {
+  const _set = cache.store.set;
+
+  cache.store.set = async (...args) => {
+    await _setKeys(cache, [args[0]]);
+    await _set.apply(cache.store, [...args]);
+  };
+
+  cache.set = cache.store.set.bind(cache.store);
+};
+
+/**
+ * Overrides given redisCache's mset() method.
+ * @param {*} cache 
+ */
+const overrideMSet = (cache) => {
+  const _mset = cache.store.mset;
+
+  cache.store.mset = async (...args) => {
+    let keys = [];
+    let _args = [...args];
```

Comment:
Extracting `keys` from the `args` should be moved to another private method for the better readability.